### DISCUSSION TITLE
Exception: Control must be added to the page first.

### DISCUSSION THREAD

**Discussion by Isnaeni2003** [flet-dev/flet#705] [General]
i have a problem.  why every call update() to change state .  but the variable you want to change always has to be above it .  it makes me difficult in the logic of writing code like that .  and i think it is a bug of flet .

I'm here to make a simple GestureDetector animation.  the case is when you scroll on the yellow container, the animated Switch runs. And can you run the animation without you having to click on a certain button or event.  for example on the web.  You can see an animation automatically when the page has just been opened.  is there a way?

this my code :
```
from flet import *


def main(page:Page):

    def scrollme(e):
        print("scroll me")
        animated.content = c2 if animated.content == c else c2
        animated.update()


    gd = GestureDetector(
        on_scroll=scrollme
        )
    c = Container(
        gd,
        bgcolor="orange",
        width=400,
        height=500
        )
    c2 = Container(
        Text("Bye!", size=50),
        alignment=alignment.center,
        width=200,
        height=200,
        bgcolor=colors.YELLOW,
    )
    animated = AnimatedSwitcher(
        c,
        transition=AnimatedSwitcherTransition.SCALE,
        duration=500,
        reverse_duration=100,
        switch_in_curve=AnimationCurve.BOUNCE_OUT,
        switch_out_curve=AnimationCurve.BOUNCE_IN,

        )

    page.add(
        Text("nice"),
        Stack([c])
        )

flet.app(target=main)
```
**This Error**
Traceback (most recent call last):
  File "/usr/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
scroll me
    self.run()
  File "/usr/lib/python3.10/threading.py", line 953, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.10/dist-packages/flet/event_handler.py", line 16, in handler
    h(r)
  File "/home/minto/belajar/ytpro/fletpro/animationflet/main.py", line 9, in scrollme
    animated.update()
  File "/usr/local/lib/python3.10/dist-packages/flet/control.py", line 258, in update
    raise Exception("Control must be added to the page first.")
Exception: Control must be added to the page first.

**Comment by ndonkoHenri** [comment]
Hey there. I don't know if you still have this issue...
To solve your problem, replace the content of your stack from `Stack([c])` to `Stack([animated])`.
